archive/p/perl/mergesort.pl

```diff
@@ -0,0 +1,18 @@
+sub merge_sort {
+    my @x = @_;
+    return @x if @x < 2;
+    my $m = int @x / 2;
+    my @a = merge_sort(@x[0 .. $m - 1]);
+    my @b = merge_sort(@x[$m .. $#x]);
+    for (@x) {
+        $_ = !@a            ? shift @b
+           : !@b            ? shift @a
+           : $a[0] <= $b[0] ? shift @a
+           :                  shift @b;
+    }
+    @x;
+}
+ 
+my @a = (4, 65, 2, -31, 0, 99, 83, 782, 1);
```

Comment:
Any chance you could pull these from the command line? See project description: https://sample-programs.therenegadecoder.com/projects/merge-sort/